cms-contentful/src/main/java/com/commercetools/sunrise/cms/contentful/ContentfulCmsService.java

```diff
@@ -7,98 +7,181 @@
      * @param locales for the localized content inside the page
-     * @return a {@code CompletionStage} containing the page content identified by the key,
+     * @return a {@link CompletionStage} containing the page content identified by the key,
      * or absent if it could not be found, or a {@link CmsServiceException} if there was a problem
-     * when obtaining the content from Contentful.
+     * when obtaining content
      */
     @Override
     public CompletionStage<Optional<CmsPage>> page(final String pageKey, final List<Locale> locales) {
-        return getEntry(pageKey, locales)
-                .thenApply(cdaEntryOptional -> cdaEntryOptional
-                        .map(cdaEntry ->
-                                new ContentfulCmsPage(cdaEntry, locales)));
+        return fetchEntry(pageKey, getLocaleForContentful(locales))
+                .thenApply(cdaEntry -> cdaEntry.map(ContentfulCmsPage::new));
     }
 
-    private CompletionStage<Optional<CDAEntry>> getEntry(final String pageKey, List<Locale> locales) {
-        return fetchEntry(pageKey, locales).thenApply(cdaArray -> Optional.ofNullable(cdaArray)
-                .filter(e -> e.items() != null && !e.items().isEmpty())
-                .map(e -> (CDAEntry) e.items().get(0)));
+    /**
+     * Convert first of provided locales to a string expected by Contentful. If list is empty return asterisk: '*'
+     * which will make request independent of locale.
+     * <p>
+     * Contentful provides only single locale to be set per request.
+     *
+     * @param locales list of locales with only first one relevant for Contentful
+     * @return string representation of requested locale adjusted for Contentful
+     */
+    private String getLocaleForContentful(List<Locale> locales) {
+        return locales.isEmpty() ? "*" : locales.get(0).toLanguageTag();
     }
 
-    private CompletionStage<CDAArray> fetchEntry(final String pageKey, List<Locale> locales) {
-        final CompletableFuture<CDAArray> future = new CompletableFuture<>();
-        final CDACallback<CDAArray> callback = new CDACallback<CDAArray>(){
+    private CompletableFuture<Optional<CDAEntry>> fetchEntry(final String pageKey, final String locale) {
+        return new ContentCallback(pageKey, locale).fetch();
+    }
+
+    /**
+     * Create new instance of {@link ContentfulCmsService} based on Contentful account credentials.
+     *
+     * @param spaceId          Contentful space ID
+     * @param token            access token to given space
+     * @param pageType         Contentful model's page type to be queried against
+     * @param pageQueryField   pageType field against which query will be run
+     * @param callbackExecutor defines execution context in which requests are executed
+     * @return instance of this service ready to serve content based on given configuration
+     */
+    public static ContentfulCmsService of(String spaceId, String token, String pageType, String pageQueryField,
```

Comment:
Overall I would suggest to use the `final` modifier for all method parameters. And you should be able to add them automatically with your IDE.